### PR TITLE
Mock Api Endpoints

### LINKED ISSUE #632: Add mock responses for endpoints
From stand-up discussion today ...

After the demo showing the `/edits` endpoint working with the UI we agreed that there were only a few endpoints remaining and that we were at a point where we could start to take care of https://github.com/cfpb/hmda-platform-ui/issues/178, removing the mock API from the front-end. To help with moving that forward we can "mock" the remaining few endpoints on the back-end.

_This will not complete those endpoints, but will allow us to see responses from them. It also doesn't include all endpoints, just enough to get the front-end to a better place._

- [ ] `/institutions/<institution>/filings/<period>/submissions/<submissionId>/irs` `GET` [example](https://github.com/cfpb/hmda-platform-ui/blob/master/server/json/irs.json)
- [ ] `/institutions/<institution>/filings/<period>/submissions/<submissionId>/irs` `POST`
    - accepts true or false
    ```json
    {"verified": true}
    ```
    - returns ([example](https://github.com/cfpb/hmda-platform-ui/blob/master/server/json/receipt.json))
    ```json
    {
        "timestamp": <timestamp>,
        "receipt": <receipt-hash>
    }
    ``` 
- [ ] `/institutions/<institution>/filings/<period>/submissions/<submissionId>/sign` `GET`
    - returns a [receipt](https://github.com/cfpb/hmda-platform-ui/blob/master/server/json/receipt.json) (nulls if not signed)
- [ ] `/institutions/<institution>/filings/<period>/submissions/<submissionId>/sign` `POST`
    - accepts true or false
    ```json
    {"signed": true}
    ```
    - returns ([example](https://github.com/cfpb/hmda-platform-ui/blob/master/server/json/receipt.json))
    ```json
    {
        "timestamp": <timestamp>,
        "receipt": <receipt-hash>
    }
    ``` 
- [ ] `/institutions/<institution>/filings/<period>/submissions/<submissionId>/summary` `GET` [example](https://github.com/cfpb/hmda-platform-ui/blob/master/server/json/summary.json)

### PR DESCRIPTION
**Pull request by nickgrippin**
Closes #632

### CODE REVIEW

**Comment by awolfe76** [comment]
@nickgrippin - can we update the `API.md ` to show these too? Maybe even call out the endpoints that are static for now.

**Comment by nickgrippin** [comment]
@awolfe76: Yeah, I'll get to that soon 😄

**Comment by awolfe76** [comment]
👍  Docs looks good. Thanks!

File: api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionSummaryPaths.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  implicit val timeout: Timeout
+
+  // institutions/<institutionId>/filings/<period>/submissions/<submissionId>/summary
+  def submissionSummaryPath(institutionId: String) =
+    path("filings" / Segment / "submissions" / IntNumber / "summary") { (period, submissionId) =>
+      extractExecutionContext { executor =>
+        timedGet { uri =>
+          implicit val ec: ExecutionContext = executor
+          val supervisor = system.actorSelection("/user/supervisor")
```
Comment (jmarin) [inline api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionSummaryPaths.scala]: It looks like we don't need the `supervisor` for these yet, until they are implemented for real. I'm ok with leaving it if this was intentional, as we will most likely use it down the road.

**Comment by jmarin** [comment]
It looks like there are some merge conflicts with this PR now. I have one comment, in general this LGTM

File: api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionSummaryPaths.scala (hunk 1 above)
Comment (nickgrippin) [inline api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionSummaryPaths.scala]: I had left some code inside the paths intentionally, with ease of expansion in mind (the refactoring of `SubmissionPaths` was done for the same reason).  But it'd be easy to remove it now and add it back in later, if that would make the PR cleaner.

**Comment by kgudel** [comment]
Other than adding some documentation this LGTM!

File: api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionSummaryPaths.scala (hunk 1 above)
Comment (jmarin) [inline api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionSummaryPaths.scala]: I'm good with leaving it as is

**Comment by nickgrippin** [comment]
@kgudel: your suggested documentation changes should be added now

**Pull request merged**